lib/src/fsmonitor.rs

```diff
@@ -0,0 +1,113 @@
+//! Interfaces with a filesystem monitor tool (currently Watchman) to
+//! efficiently query for filesystem updates, without having to crawl the entire
+//! repository. This is particularly useful for large repositories, or for
+//! repositories which are virtualized or reside on a network store.
```

Comment:
> nit: Replace "store" by "file system" if that's what you mean.

Also, what do you mean by "virtualized" repositories?

FYI, what we plan to do internally is to ask our VFS what the changes are compared to its base commit. Since it knows what the base commit is, we should normally get a pretty small list of files to update the TreeBuilder with.

@martinvonz Watchman has an API to hook into a VFS and query it for changes since a given commit!